storage/watcher.go

```diff
@@ -113,3 +124,14 @@ func (ws *watchStream) Close() {
 	close(ws.ch)
 	watchStreamGauge.Dec()
 }
+
+func (ws *watchStream) Rev() int64 {
+	ws.mu.Lock()
+	defer ws.mu.Unlock()
+
+	s, ok := ws.watchable.(*watchableStore)
+	if !ok {
+		return 0
```

Comment:
can you add Rev() to the interface of watchable, then you do not need to do the type assertion.